src/stk500.c

```diff
@@ -1415,7 +1454,19 @@ static void stk500_print_parms1(const PROGRAMMER *pgm, const char *p, FILE *fp)
   }
 
   stk500_getparm(pgm, Parm_STK_SCK_DURATION, &sck_duration);
-  fmsg_out(fp, "%sSCK period      : %.1f us\n", p, sck_duration * 8.0e6 / STK500_XTAL + 0.05);
+  fmsg_out(fp, "%sSCK period      : %.1f us\n", p, sck_duration * 8.0e6 / PDATA(pgm)->xtal + 0.0499);
+
+  const char *unit;
+  double f = PDATA(pgm)->xtal;
+  if (f >= 1e6) {
+    f /= 1e6;
+    unit = "MHz";
+  } else if (f >= 1e3) {
+    f /= 1000;
+    unit = "kHz";
+  } else
+    unit = "Hz";
+  fmsg_out(fp, "%sXTAL frequency  : %.3f %s\n", p, f, unit);
```

Comment:
Is .3 sufficient? Might .5 or .6 be better?

Reply:
Yes, I changed to variable decimal count 6 for MHz, 3 for kHz, 0 for Hz